tests/components/suez_water/test_init.py

```diff
@@ -35,3 +38,78 @@ async def test_initialization_setup_api_error(
     await setup_integration(hass, mock_config_entry)
 
     assert mock_config_entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_migration_version_rollback(
+    hass: HomeAssistant,
+    suez_client: AsyncMock,
+) -> None:
+    """Test that a version rollback does not impact config."""
+    future_entry = MockConfigEntry(
+        unique_id=MOCK_DATA[CONF_COUNTER_ID],
+        domain=DOMAIN,
+        title="Suez mock device",
+        data=MOCK_DATA,
+        version=3,
+    )
+    await setup_integration(hass, future_entry)
+    assert future_entry.state is ConfigEntryState.MIGRATION_ERROR
+
+
+async def test_no_migration_current_version(
+    hass: HomeAssistant,
+    suez_client: AsyncMock,
+) -> None:
+    """Test that a current version does not migrate."""
+    current_entry = MockConfigEntry(
+        unique_id=MOCK_DATA[CONF_COUNTER_ID],
+        domain=DOMAIN,
+        title="Suez mock device",
+        data=MOCK_DATA,
+        version=2,
+    )
+    await setup_integration(hass, current_entry)
+    assert current_entry.state is ConfigEntryState.LOADED
+    assert current_entry.unique_id == MOCK_DATA[CONF_COUNTER_ID]
+
+
+async def test_migration_version_1_to_2(
+    hass: HomeAssistant,
+    suez_client: AsyncMock,
+) -> None:
+    """Test that a migration from 1 to 2 change unique_id."""
+    past_entry = MockConfigEntry(
+        unique_id=MOCK_DATA[CONF_USERNAME],
+        domain=DOMAIN,
+        title=MOCK_DATA[CONF_USERNAME],
+        data=MOCK_DATA,
+        version=1,
+        minor_version=0,
+    )
+
+    await setup_integration(hass, past_entry)
+    assert past_entry.state is ConfigEntryState.LOADED
+    assert past_entry.unique_id == MOCK_DATA[CONF_COUNTER_ID]
+    assert past_entry.title == MOCK_DATA[CONF_USERNAME]
+
+
+async def test_migration_version_1_to_2_int_counter(
```

Comment:
I think we should remove this test as the schema I guess has always been a string so why would it suddenly be an int?